app/coffee/WebsocketController.coffee

```diff
@@ -15,11 +15,18 @@ module.exports = WebsocketController =
 	PROTOCOL_VERSION: 2
 
 	joinProject: (client, user, project_id, callback = (error, project, privilegeLevel, protocolVersion) ->) ->
+		if not client.connected
+			metrics.inc('disconnected_join_project')
+			return callback()
```

Comment:
To match the other metrics how about `metrics.inc('editor.join-project.disconnected', 1, {status: 'at-start'})` and below `metrics.inc('editor.join-project.disconnected', 1, {status: 'after-web-api-call'})`?  Similarly for the `joinDoc` method.